src/main/kotlin/org/rust/lang/core/macros/ExpandedMacroStorage.kt

```diff
@@ -652,3 +663,34 @@ private fun calcStubIndex(psi: StubBasedPsiElement<*>): Int {
 
 private val VirtualFile.fileId: Int
     get() = (this as VirtualFileWithId).id
+
+// Use WeakReference it's relatively easy to load RangeMap again
```

Comment:
Do you mean that `VirtualFile#loadRangeMap` is quite cheap?
If yes, let's rewrite it somehow to make it clearer

Reply:
Yes. Done